MekHQ/src/mekhq/gui/adapter/FinanceTableMouseAdapter.java

```diff
@@ -36,18 +34,22 @@
 import mekhq.gui.dialog.EditTransactionDialog;
 import mekhq.gui.model.FinanceTableModel;
 
-public class FinanceTableMouseAdapter extends MouseInputAdapter implements ActionListener {
+public class FinanceTableMouseAdapter extends JPopupMenuAdapter {
     private CampaignGUI gui;
     private JTable financeTable;
     private FinanceTableModel financeModel;
 
-    public FinanceTableMouseAdapter(CampaignGUI gui, JTable financeTable, FinanceTableModel financeModel) {
-        super();
+    protected FinanceTableMouseAdapter(CampaignGUI gui, JTable financeTable, FinanceTableModel financeModel) {
         this.gui = gui;
         this.financeTable = financeTable;
         this.financeModel = financeModel;
     }
 
+    public static void connect(CampaignGUI gui, JTable financeTable, FinanceTableModel financeModel) {
```

Comment:
Could this being static cause issues with your goal of testing these components?

Reply:
No, I don't really have a good way to test Swing components unfortunately.